src/libs/Network/SequentialQueue.js

```diff
@@ -114,9 +114,21 @@ function getCurrentRequest() {
     return currentRequest;
 }
 
+/**
+ * Returns a promise that resolves when the last request on the queue responds.
+ * @returns {Promise}
+ */
+function getIsReadyPromise() {
+    if (!isSequentialQueueRunning) {
+        return Promise.resolve();
+    }
```

Comment:
**question:** Why do we use the Boolean instead of the `isReadyPromise` itself? If the queue is not running then the promise should be resolved, no?

Reply:
sorry I misread this during our 1:1 earlier, what you're describing was actually what I had initially but [broke a bunch of jest unit tests](https://github.com/Expensify/App/actions/runs/3364570445/jobs/5579073938), turns out the promise gets reset when the queue stops running so `read` commands that don't need to wait on the sequential queue don't run, therefore we have to use the Boolean here.

Reply:
What about them broke?

>the promise gets reset when the queue stops running so the promise just never resolves

I see it resolving here: 

https://github.com/Expensify/App/blob/f17042cb6c2dd3fe4b738d7406c8a818e39483c1/src/libs/Network/SequentialQueue.js#L69

And getting reset when the queue starts again here:

https://github.com/Expensify/App/blob/f17042cb6c2dd3fe4b738d7406c8a818e39483c1/src/libs/Network/SequentialQueue.js#L57-L59

Should work I think? Lemme see if I can help figure out what is happening...
Doesn't seem like it would be possible for that promise to be replaced

Reply:
The unit tests are passing for me locally so I'm not able to repro. Did you have some other changes that maybe could have broken the tests for that run? Or maybe they flaked for some reason.

Reply:
Hmm, let me try this again, the last time I did this was yesterday and I don't quite remember what happened.

Reply:
Huh, they're passing for me now locally, sorry to make you look into that, just pushed these changes!

Reply:
Huh, they're passing for me now locally, thanks for looking into that for me, just pushed these changes!